### ISSUE TITLE
Elastic Beanstalk can't find a platform version that matches "Docker running on 64bit Amazon Linux 2"

### ISSUE THREAD

**Issue by vlucas** [aws/aws-elastic-beanstalk-cli#28]
I run:

### Description

Briefly describe the bug you are facing.

### Steps to reproduce

1. Setup a new git-based project with a Dockerfile in it
2. Run:
```
eb init
```

I selected:
- region `us-east-2` (Ohio) (14)
- Default Docker image (1)

I then try to setup an environment:

3. Run:
```
> eb create prod

ERROR: NotFoundError - Elastic Beanstalk can't find a platform version that matches "Docker running on 64bit Amazon Linux 2".
```

I cannot get past this and cannot deploy my app.

### Observed result

Please provide command output with `--debug` flag set.

```
2020-05-05 11:29:55,622 (DEBUG) cement.core.foundation : laying cement for the 'eb' application
2020-05-05 11:29:55,622 (DEBUG) cement.core.hook : defining hook 'pre_setup'
2020-05-05 11:29:55,622 (DEBUG) cement.core.hook : defining hook 'post_setup'
2020-05-05 11:29:55,622 (DEBUG) cement.core.hook : defining hook 'pre_run'
2020-05-05 11:29:55,622 (DEBUG) cement.core.hook : defining hook 'post_run'
2020-05-05 11:29:55,623 (DEBUG) cement.core.hook : defining hook 'pre_argument_parsing'
2020-05-05 11:29:55,623 (DEBUG) cement.core.hook : defining hook 'post_argument_parsing'
2020-05-05 11:29:55,623 (DEBUG) cement.core.hook : defining hook 'pre_close'
2020-05-05 11:29:55,623 (DEBUG) cement.core.hook : defining hook 'post_close'
2020-05-05 11:29:55,623 (DEBUG) cement.core.hook : defining hook 'signal'
2020-05-05 11:29:55,623 (DEBUG) cement.core.hook : defining hook 'pre_render'
2020-05-05 11:29:55,623 (DEBUG) cement.core.hook : defining hook 'post_render'
2020-05-05 11:29:55,623 (DEBUG) cement.core.hook : registering hook 'add_handler_override_options' from cement.core.foundation into hooks['post_setup']
2020-05-05 11:29:55,623 (DEBUG) cement.core.hook : registering hook 'handler_override' from cement.core.foundation into hooks['post_argument_parsing']
2020-05-05 11:29:55,623 (DEBUG) cement.core.handler : defining handler type 'extension' (IExtension)
2020-05-05 11:29:55,623 (DEBUG) cement.core.handler : defining handler type 'log' (ILog)
2020-05-05 11:29:55,623 (DEBUG) cement.core.handler : defining handler type 'config' (IConfig)
2020-05-05 11:29:55,623 (DEBUG) cement.core.handler : defining handler type 'mail' (IMail)
2020-05-05 11:29:55,623 (DEBUG) cement.core.handler : defining handler type 'plugin' (IPlugin)
2020-05-05 11:29:55,624 (DEBUG) cement.core.handler : defining handler type 'output' (IOutput)
2020-05-05 11:29:55,624 (DEBUG) cement.core.handler : defining handler type 'argument' (IArgument)
2020-05-05 11:29:55,624 (DEBUG) cement.core.handler : defining handler type 'controller' (IController)
2020-05-05 11:29:55,624 (DEBUG) cement.core.handler : defining handler type 'cache' (ICache)
2020-05-05 11:29:55,624 (DEBUG) cement.core.handler : registering handler '<class 'cement.core.extension.CementExtensionHandler'>' into handlers['extension']['cement']
2020-05-05 11:29:55,642 (DEBUG) cement.ext.ext_plugin : plugin config dir /etc/eb/plugins.d does not exist.
2020-05-05 11:29:55,642 (DEBUG) cement.ext.ext_plugin : plugin config dir /Users/vlucas/.eb/plugins.d does not exist.
2020-05-05 11:29:55,657 (DEBUG) ebcli.core.hooks : -- EBCLI Version: 3.18.1
2020-05-05 11:29:55,658 (DEBUG) ebcli.core.hooks : -- Python Version: 3.7.5 (default, Nov  1 2019, 02:16:38)
[Clang 10.0.0 (clang-1000.11.45.5)]
2020-05-05 11:29:55,658 (DEBUG) ebcli.core.fileoperations : Project root found at: /Users/vlucas/Code/projects/seamless.cloud
2020-05-05 11:29:55,662 (DEBUG) ebcli.objects.sourcecontrol : Getting current branch name by performing `git rev-parse --abbrev-ref HEAD`
2020-05-05 11:29:55,674 (DEBUG) ebcli.objects.sourcecontrol : master
2020-05-05 11:29:55,674 (DEBUG) ebcli.core.fileoperations : Project root found at: /Users/vlucas/Code/projects/seamless.cloud
2020-05-05 11:29:55,678 (DEBUG) ebcli.core.fileoperations : Project root found at: /Users/vlucas/Code/projects/seamless.cloud
2020-05-05 11:29:55,684 (DEBUG) ebcli.core.fileoperations : Project root found at: /Users/vlucas/Code/projects/seamless.cloud
2020-05-05 11:29:55,688 (DEBUG) ebcli.objects.sourcecontrol : Getting current branch name by performing `git rev-parse --abbrev-ref HEAD`
2020-05-05 11:29:55,702 (DEBUG) ebcli.objects.sourcecontrol : master
2020-05-05 11:29:55,703 (DEBUG) ebcli.core.fileoperations : Project root found at: /Users/vlucas/Code/projects/seamless.cloud
2020-05-05 11:29:55,707 (DEBUG) ebcli.core.fileoperations : Project root found at: /Users/vlucas/Code/projects/seamless.cloud
2020-05-05 11:29:55,712 (DEBUG) ebcli.core.fileoperations : Project root found at: /Users/vlucas/Code/projects/seamless.cloud
2020-05-05 11:29:55,717 (DEBUG) ebcli.core.fileoperations : Project root found at: /Users/vlucas/Code/projects/seamless.cloud
2020-05-05 11:29:55,721 (DEBUG) ebcli.core.fileoperations : Project root found at: /Users/vlucas/Code/projects/seamless.cloud
2020-05-05 11:29:55,724 (DEBUG) ebcli.objects.sourcecontrol : Getting current branch name by performing `git rev-parse --abbrev-ref HEAD`
2020-05-05 11:29:55,736 (DEBUG) ebcli.objects.sourcecontrol : master
2020-05-05 11:29:55,736 (DEBUG) ebcli.core.fileoperations : Project root found at: /Users/vlucas/Code/projects/seamless.cloud
2020-05-05 11:29:55,740 (DEBUG) ebcli.core.fileoperations : Project root found at: /Users/vlucas/Code/projects/seamless.cloud
2020-05-05 11:29:55,744 (DEBUG) ebcli.lib.elasticbeanstalk : Inside list_platform_branches api wrapper
2020-05-05 11:29:55,745 (DEBUG) ebcli.lib.aws : Creating new Botocore Session
2020-05-05 11:29:55,745 (DEBUG) ebcli.lib.aws : Botocore version: 1.15.49
2020-05-05 11:29:55,764 (DEBUG) ebcli.lib.aws : Creating new Botocore Client for elasticbeanstalk
2020-05-05 11:29:55,817 (DEBUG) ebcli.lib.aws : Successfully created session for elasticbeanstalk
2020-05-05 11:29:55,817 (DEBUG) ebcli.lib.aws : Making api call: (elasticbeanstalk, list_platform_branches) to region: us-east-2 with args:{'Filters': [{'Attribute': 'BranchName', 'Operator': '=', 'Values': ['Docker running on 64bit Amazon Linux 2']}]}
2020-05-05 11:29:56,098 (DEBUG) ebcli.lib.aws : API call finished, status = 200
2020-05-05 11:29:56,098 (DEBUG) ebcli.lib.aws : Response: {'PlatformBranchSummaryList': [{'PlatformName': 'Docker', 'BranchName': 'Docker running on 64bit Amazon Linux 2', 'BranchOrder': '11000', 'LifecycleState': 'Supported', 'SupportedTierList': ['WebServer/Standard', 'Worker/SQS/HTTP']}], 'ResponseMetadata': {'RequestId': 'e9ce65a1-3725-44c1-9222-7970f1b7510b', 'HTTPStatusCode': 200, 'date': 'Tue, 05 May 2020 16:29:56 GMT', 'RetryAttempts': 0}}
2020-05-05 11:29:56,098 (DEBUG) ebcli.lib.elasticbeanstalk : Inside list_platform_versions api wrapper
2020-05-05 11:29:56,098 (DEBUG) ebcli.lib.aws : Making api call: (elasticbeanstalk, list_platform_versions) to region: us-east-2 with args:{'Filters': [{'Type': 'PlatformBranchName', 'Operator': '=', 'Values': ['Docker running on 64bit Amazon Linux 2']}]}
2020-05-05 11:29:56,280 (DEBUG) ebcli.lib.aws : API call finished, status = 200
2020-05-05 11:29:56,281 (DEBUG) ebcli.lib.aws : Response: {'PlatformSummaryList': [], 'ResponseMetadata': {'RequestId': 'db9d58fb-7113-4282-9601-8065fec76580', 'HTTPStatusCode': 200, 'date': 'Tue, 05 May 2020 16:29:56 GMT', 'RetryAttempts': 0}}
2020-05-05 11:29:56,283 (INFO) eb : Traceback (most recent call last):
  File "/Users/vlucas/Library/Python/3.7/lib/python/site-packages/ebcli/core/ebrun.py", line 62, in run_app
    app.run()
  File "/Users/vlucas/Library/Python/3.7/lib/python/site-packages/cement/core/foundation.py", line 797, in run
    return_val = self.controller._dispatch()
  File "/Users/vlucas/Library/Python/3.7/lib/python/site-packages/cement/core/controller.py", line 472, in _dispatch
    return func()
  File "/Users/vlucas/Library/Python/3.7/lib/python/site-packages/cement/core/controller.py", line 478, in _dispatch
    return func()
  File "/Users/vlucas/Library/Python/3.7/lib/python/site-packages/ebcli/core/abstractcontroller.py", line 92, in default
    self.do_command()
  File "/Users/vlucas/Library/Python/3.7/lib/python/site-packages/ebcli/controllers/create.py", line 213, in do_command
    platform = _determine_platform(platform, iprofile)
  File "/Users/vlucas/Library/Python/3.7/lib/python/site-packages/ebcli/controllers/create.py", line 587, in _determine_platform
    platform_string)
  File "/Users/vlucas/Library/Python/3.7/lib/python/site-packages/ebcli/operations/platformops.py", line 189, in get_platform_for_platform_string
    return platform_version_ops.get_preferred_platform_version_for_branch(platform_string)
  File "/Users/vlucas/Library/Python/3.7/lib/python/site-packages/ebcli/operations/platform_version_ops.py", line 348, in get_preferred_platform_version_for_branch
    branch_name))
ebcli.objects.exceptions.NotFoundError: Elastic Beanstalk can't find a platform version that matches "Docker running on 64bit Amazon Linux 2".
```

### Expected result

I expected this to work.

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: Mac OSX

`Add --debug flag to command you are running`

**Comment by halcarleton** [comment]
Thanks for reporting this @vlucas,

This appears to be a data issue, not a bug in the EBCLI. The issue is that the platform/platform branch that you selected has no platform versions associated with it.

I will report this issue internally, but I would recommend that you also post to the AWS Elastic Beanstalk forums.

In the meantime you should be able to unblock yourself by selecting a different platform.

**Comment by RealRobin-com** [comment]
Having a similar issue when trying to run `eb config put test-sc.cfg.yml`

`ERROR: NotFoundError - Elastic Beanstalk can't find a platform version that matches "Ruby 2.6 running on 64bit Amazon Linux 2".`

**Comment by halcarleton** [comment]
> Having a similar issue when trying to run `eb config put test-sc.cfg.yml`

@RealRobin-com Please see the following: https://github.com/aws/aws-elastic-beanstalk-cli/issues/32

**Comment by EGSatelix** [comment]
> Thanks for reporting this @vlucas,
> 
> This appears to be a data issue, not a bug in the EBCLI. The issue is that the platform/platform branch that you selected has no platform versions associated with it.
> 
> I will report this issue internally, but I would recommend that you also post to the [AWS Elastic Beanstalk forums](https://forums.aws.amazon.com/forum.jspa?forumID=86).
> 
> In the meantime you should be able to unblock yourself by selecting a different platform.


Hi,
I have a similary error but i can't select another platform, button is disabled.

> ERROR: InvalidParameterValueError - Platform 'arn:aws:elasticbeanstalk:eu-west-3::platform/Docker running on 64bit Amazon Linux/2.12.17' does not exist.

**Comment by halcarleton** [comment]
> Hi,
> I have a similary error but i can't select another platform, button is disabled.
> 
> > ERROR: InvalidParameterValueError - Platform 'arn:aws:elasticbeanstalk:eu-west-3::platform/Docker running on 64bit Amazon Linux/2.12.17' does not exist.

Hey @EGSatelix, please open a new issue and fill out all the requested information.
Thanks

**Comment by DarwinJS** [comment]
I feel this issue needs to be reopened.

I get the same error - however - I get it with a configuration file saved from the exact eb application I am trying to apply updates to.  'eb config save' works fine.

I am running eb cli  3.19.3 in an amazon linux 2 container on MacOS - using volumes to pass through to my credentials file (not that it should matter)

**Comment by bittersweet** [comment]
We also had the error `ERROR: NotFoundError - Elastic Beanstalk can't find a platform version that matches "Docker running on 64bit Amazon Linux"` pop up with version `EB CLI 3.19.4 (Python 3.9.5)`.

In our case, it turend out the access credentials were missing the `elasticbeanstalk:ListPlatformVersions` permission.

**Comment by adonig** [comment]
I know this issue is already closed but I have the feeling that my problems are related.

I was trying to deploy a Python/Django app using Docker. I initialized the app using `eb init` and chose Docker as platform.

When trying to create an environment I ran into a strange "invalid option specification" problem.

```
Printing Status:
2022-02-16 12:53:21    INFO    createEnvironment is starting.
2022-02-16 12:53:23    INFO    Using elasticbeanstalk-eu-central-1-_redacted_ as Amazon S3 storage bucket for environment data.
2022-02-16 12:53:24    ERROR   "option_settings" in one of the configuration files failed validation. More details to follow.
2022-02-16 12:53:24    ERROR   Invalid option specification (Namespace: 'aws:elasticbeanstalk:container:python:staticfiles', OptionName: '/static/'): Unknown configuration setting.
2022-02-16 12:53:24    ERROR   Failed to launch environment.
```

I tried to fix it with an `.ebextensions` config file, but it didn't work. It looks like `eb create` was using the wrong platform (python instead of docker).

I tried to select the platform with `eb create --platform docker` and ran into the following error:

```
ERROR: InvalidParameterValueError - Platform ARN is invalid: Not an IAM ARN: 64bit Amazon Linux 2 v3.4.11 running Docker.
```

I tried to select the default platform, but it didn't help. When I ran `eb platform show` I got the error:

```
$ eb platform show
ERROR: NotFoundError - Elastic Beanstalk can't find a platform version that matches "Docker running on 64bit Amazon Linux 2".
```

I thought that my awsebcli might be deprecated, but I'm running `EB CLI 3.20.3 (Python 3.10.)` installed with homebrew on OS X.

Some people seemed to have a permission problem but that's unlikely because I use an IAM account with full admin permissions.
